File: app/src/main/java/social/entourage/android/map/MapEntourageFragmentLifecycleCallbacks.java
package social.entourage.android.map;

import android.content.Context;
import androidx.annotation.NonNull;
import androidx.fragment.app.DialogFragment;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;

import java.util.ArrayList;

/**
 * Created by Mihai Ionescu on 10/10/2017.
 */

public class MapEntourageFragmentLifecycleCallbacks extends FragmentManager.FragmentLifecycleCallbacks {

    private ArrayList<Fragment> fragmentList = new ArrayList<>();

    @Override
    public void onFragmentAttached(@NonNull final FragmentManager fm, @NonNull final Fragment f, @NonNull final Context context) {
        super.onFragmentAttached(fm, f, context);
        if (f != null) {
            fragmentList.add(f);
        }
    }

    @Override
    public void onFragmentDetached(@NonNull final FragmentManager fm, @NonNull final Fragment f) {
        super.onFragmentDetached(fm, f);
        if (f != null) {
            fragmentList.remove(f);
        }
    }

    public Fragment getTopFragment() {
        if (fragmentList.size() > 0) {
            return fragmentList.get(fragmentList.size() - 1);
        }
        return null;
    }

    public void dismissAllDialogs() {
        int count = fragmentList.size() - 1;
        while (count >= 0) {
            Fragment fragment = fragmentList.get(count);
            if (fragment instanceof DialogFragment) {
                ((DialogFragment) fragment).dismissAllowingStateLoss();
            }
            count--;
        }
    }

}

File: app/src/main/java/social/entourage/android/map/MapEntourageFragmentLifecycleCallbacks.kt
package social.entourage.android.map

import android.content.Context
import androidx.fragment.app.DialogFragment
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import social.entourage.android.EntourageEvents

import java.util.ArrayList

/**
 * Created by Mihai Ionescu on 10/10/2017.
 */

class MapEntourageFragmentLifecycleCallbacks : FragmentManager.FragmentLifecycleCallbacks() {

    private val fragmentList = ArrayList<Fragment>()

    val topFragment: Fragment?
        get() = if (fragmentList.size > 0) {
            fragmentList[fragmentList.size - 1]
        } else null

    override fun onFragmentAttached(fm: FragmentManager, f: Fragment, context: Context) {
        super.onFragmentAttached(fm, f, context)
        fragmentList.add(f)
    }

    override fun onFragmentDetached(fm: FragmentManager, f: Fragment) {
        super.onFragmentDetached(fm, f)
        fragmentList.remove(f)
    }

    fun dismissAllDialogs() {
        var count = fragmentList.size - 1
        while (count >= 0) {
            val fragment = fragmentList[count]
            try{
                if (fragment is DialogFragment) {
                    fragment.dismiss()
                }
            } catch(e: IllegalStateException) {
                EntourageEvents.logEvent(EntourageEvents.EVENT_ILLEGAL_STATE)
            }
            count--
        }
    }

}
